File: Csharp/CodeWars/src/Resolutions/LongestVowelChain.cs
namespace CodeWars.src.Resolutions;

public class LongestVowelChain
{
    public static int Solve(string str)
    {
        string vowels = "aeiou";
        int result = 0;
        int count = 0;

        foreach (var item in str.ToLower())
        {
            if (vowels.Contains(item))
                count++;
            else
            {
                result = Math.Max(result, count);
                count = 0;
            }
        }

        return Math.Max(result, count);
    }
}

File: TypeScript/Codewars/src/resolutions/Longest-vowel-chain.ts
export function solve(s: string) {
	const vowels = ["a", "e", "i", "o", "u"];
	const result: number[] = [];
	let count = 0;

	for (const element of s) {
		if (vowels.includes(element)) count++;
		else {
			result.push(count);
			count = 0;
		}
	}

	return Math.max(...result);
}

console.log(solve("iiihoovaeaaaoougjyaw"));
